MoreLinq/FullGroupJoin.cs

```diff
@@ -76,7 +76,7 @@ static partial class MoreEnumerable
             IEnumerable<TSecond> second,
             Func<TFirst, TKey> firstKeySelector,
             Func<TSecond, TKey> secondKeySelector,
-            IEqualityComparer<TKey> comparer)
+            IEqualityComparer<TKey>? comparer)
```

Comment:
Nullable is not enabled for this file, I don't believe. Maybe should enable before finishing PR?

Reply:
Nullable is enabled for the whole project in `.csproj`, unless I'm missing something? As far as I can tell it's enabled by default for the project and disabled on a file by file basis for certain files (like `Lookup.cs`).

Reply:
Oh. Sure enough. I was remembering reading the `#nullable enable` in #805, but I see now that's just in the `MoreLinq.Tests` project, and not the main project. I'm good then! :)

Reply:
Like in #805, how about adding `#nullable enable` in `FullGroupJoinTest.cs` and changing the following line in the test code:

https://github.com/morelinq/MoreLINQ/blob/7458bf13c3e47490abfccdb09f18284654a371dd/MoreLinq.Test/FullGroupJoinTest.cs#L138

to:

```c#
                    return listA.FullGroupJoin(listB, getKey, getKey, ValueTuple.Create, comparer: null);
```

This would ensure that the project doesn't compile is someone removes the annotation by accident.